addons/event/models/event_registration.py

```diff
@@ -148,12 +149,53 @@ def _synchronize_partner_values(self, partner, fnames=None):
 
     @api.model_create_multi
     def create(self, vals_list):
+        for vals in vals_list:
+            vals['utm_campaign_id'] = vals.get('utm_campaign_id') or self.get_utm_params(['utm_campaign_id']).get('utm_campaign_id')
+            vals['utm_source_id'] = vals.get('utm_source_id') or self.get_utm_params(['utm_source_id']).get('utm_source_id')
+            vals['utm_medium_id'] = vals.get('utm_medium_id') or self.get_utm_params(['utm_medium_id']).get('utm_medium_id')
+
         registrations = super(EventRegistration, self).create(vals_list)
         if registrations._check_auto_confirmation():
             registrations.sudo().action_confirm()
 
         return registrations
 
+    def get_utm_params(self, fields):
+        values = {}
```

Comment:
In the default_get, can we add a short comment/docstring to specify it emulates utm.mixin?